### PR TITLE
add null checks in iface_from_ifaddrs

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

**Comment by arvidn** [comment]
do you know if `RC_2_0` is subject to this issue too?

**Comment by aldenml** [comment]
just checked and yes. Do you want me to submit the PR to `RC_2_0` instead? (expecting a yes :) )

**Comment by aldenml** [comment]
@arvidn changed the PR to target `RC_2_0`

File: src/enum_net.cpp [hunk 1]
```diff
@@ -587,12 +587,18 @@ int _System __libsocket_sysctl(int* mib, u_int namelen, void *oldp, size_t *oldl
 #if TORRENT_USE_IFADDRS && !defined TORRENT_BUILD_SIMULATOR
 	bool iface_from_ifaddrs(ifaddrs *ifa, ip_interface &rv)
 	{
+		// some android ifaddrs implementations are buggy
+		if (ifa->ifa_addr == nullptr) return false;
+
```
Comment (gubatron) [inline src/enum_net.cpp:592]: why not have the three blocks the same way?

```cpp
if (ifa->ifa_addr != nullptr)
{
        rv.interface_address = sockaddr_to_address(ifa->ifa_addr);
}
```

(this works well when I test it)

**Review by arvidn — APPROVED** [review]

**Comment by gubatron** [comment]
Please include this patch also in `RC_1_2`

**Pull request merged**